### PR TITLE
Validate the username when creating users through the api

### PR DESCRIPTION
**Pull request by felixoi**

### CODE REVIEW

**Comment by felixoi** [comment]
I have no clue why the test fail.

**Comment by lukegb** [comment]
That's not your fault, the requirements.txt is actually broken. I'll fix the tests first, and then ask you to rebase.

**Comment by felixoi** [comment]
Good. Added a test and travis is passing now.

File: spongeauth/api/views.py [hunk 1]
```diff
@@ -73,6 +74,11 @@ def _create_user(request):
     verified = request.POST.get('verified', 'false') == 'true'
     dummy = request.POST.get('dummy', 'false') == 'true'
 
+    try:
+        accounts.models.validate_username(username)
```
Comment (lukegb) [inline spongeauth/api/views.py]: I'd rather you run all model validation here: basically, before `.save` is called, you should call `.full_clean()`, which will do all the validation (including username validation):

https://docs.djangoproject.com/en/2.1/ref/models/instances/#django.db.models.Model.full_clean

Your test shouldn't need to change.

File: spongeauth/api/views.py [hunk 2]
```diff
@@ -73,6 +74,11 @@ def _create_user(request):
...
+    except ValidationError as exc:
+        return django.http.JsonResponse({'errors': exc.messages}, status=422)
```
Comment (lukegb) [inline spongeauth/api/views.py]: Instead of 422, please use `httplib.UNPROCESSABLE_ENTITY`

(you'll need to `import httplib` too)

**Review by lukegb — CHANGES_REQUESTED** [review]
Thanks for the contribution! Two small things, otherwise you're good to go.

File: spongeauth/api/views.py (hunk 2 above)
Comment (lukegb) [reply spongeauth/api/views.py]: Whoops, sorry, it's `import http` in Python 3, and `http.HTTPStatus.UNPROCESSABLE_ENTITY`

**Review by lukegb — APPROVED** [review]

**Pull request merged**